cmd/nginx-ingress/aws.go

```diff
@@ -0,0 +1,108 @@
+// +build aws
+
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"log"
+	"math/rand"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/marketplacemetering"
+	"github.com/aws/aws-sdk-go-v2/service/marketplacemetering/types"
+
+	"github.com/dgrijalva/jwt-go/v4"
+)
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-"
+
+var (
+	productCode   string
+	pubKeyVersion int32 = 1
+	pubKeyString  string
+	nonce         string
+)
+
+func init() {
+	rand.Seed(jwt.Now().UnixNano())
+	nonce = RandStringBytesRmndr(32)
+
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		log.Fatalf("Error loading AWS configuration: %v", err)
+	}
+
+	mpm := marketplacemetering.New(marketplacemetering.Options{Region: cfg.Region, Credentials: cfg.Credentials})
+
+	var notEnt *types.CustomerNotEntitledException
+	var invalidRegion *types.InvalidRegionException
+
+	out, err := mpm.RegisterUsage(context.TODO(), &marketplacemetering.RegisterUsageInput{ProductCode: &productCode, PublicKeyVersion: &pubKeyVersion, Nonce: &nonce})
+	if err != nil {
+		if errors.As(err, &notEnt) {
```

Comment:
I think errors.As is used to convert the error to a specific type, usually to get error details specific to that error type. In this case, just the type is checked so I think it should be errors.Is() ?
WDYT?

Reply:
Updated to print details